src/test/java/org/folio/dao/DataImportQueueDaoTest.java

```diff
@@ -37,35 +41,54 @@ public class DataImportQueueDaoTest  {
 
   @InjectMocks
   DataImportQueueItemDao queueItemDaoImpl = new DataImportQueueItemDaoImpl();
-  
 
   UUID storedItemUUID;
+
   @Before
   public void setUp() {
-    MockitoAnnotations.initMocks(this);
+    MockitoAnnotations.openMocks(this);
     when(postgresClientFactory.getInstance()).thenReturn(postgresClient);
   }
-  @Test 
+
+  @Test
   public void shouldAddQueueItem(TestContext context) {
     // given
-    when(postgresClient.execute(anyString(), any(Tuple.class)))
-      .thenReturn(Future.succeededFuture(new LocalRowSet(1)));
+    doAnswer((InvocationOnMock invocation) -> {
+        Promise<RowSet<Row>> promise = invocation.getArgument(2);
+        promise.complete(new LocalRowSet(1));
+        return null;
+      })
+      .when(postgresClient)
+      .execute(
+        anyString(),
+        any(Tuple.class),
+        ArgumentMatchers.<Promise<RowSet<Row>>>any()
+      );
+
     // when
-  DataImportQueueItem queueItem = new DataImportQueueItem();
-  queueItem.setFilePath("test/file/path");
-  storedItemUUID = UUID.randomUUID();
-  queueItem.setId(storedItemUUID.toString());
-  queueItem.setUploadDefinitionId(UUID.randomUUID().toString());
-  queueItem.setJobExecutionId(UUID.randomUUID().toString());
-  queueItem.setSize(1000);
-  queueItem.setOriginalSize(5000);
-  DateTime now = new DateTime();
-  queueItem.setTimestamp(now.toString());
-  queueItemDaoImpl.addQueueItem(queueItem)
-    // then
-    .onComplete(context.asyncAssertSuccess(x ->
-      verify(postgresClient).execute(anyString(), any(Tuple.class))));
-//    assert( queueItemDao.addQueueItem(queueItem).succeeded() == true);
+    DataImportQueueItem queueItem = new DataImportQueueItem();
+    queueItem.setFilePath("test/file/path");
+    storedItemUUID = UUID.randomUUID();
+    queueItem.setId(storedItemUUID.toString());
+    queueItem.setUploadDefinitionId(UUID.randomUUID().toString());
+    queueItem.setJobExecutionId(UUID.randomUUID().toString());
+    queueItem.setSize(1000);
+    queueItem.setOriginalSize(5000);
+    DateTime now = new DateTime();
+    queueItem.setTimestamp(now.toString());
+    queueItemDaoImpl
+      .addQueueItem(queueItem)
+      // then
+      .onComplete(
+        context.asyncAssertSuccess(x -> {
+          verify(postgresClient, times(1))
+            .execute(
+              anyString(),
+              any(Tuple.class),
+              ArgumentMatchers.<Promise<RowSet<Row>>>any()
+            );
+          verifyNoMoreInteractions(postgresClient);
```

Comment:
Always good to include this in your test; it will ensure mocked calls are made and that no other calls were made.  I think this might have also caught the mocking error (it will tell you the exact issue, e.g.):
```

Argument(s) are different! Wanted:
postgresClient.execute(
    <any string>,
    <any io.vertx.sqlclient.Tuple>
);
-> at org.folio.rest.persist.PostgresClient.execute(PostgresClient.java:3869)
Actual invocations have different arguments:
postgresClient.execute(
    "INSERT INTO data_import_global.queue_items (id, jobExecutionId,uploadDefinitionId,size,originalSize,filePath,timestamp) VALUES ($1, $2, $3, $4, $5, $6, $7)",
    io.vertx.sqlclient.impl.ArrayTuple@40de8f93,
    Future{result=org.folio.rest.persist.helpers.LocalRowSet@6ff0b1cc}
);
```